.github/workflows/push_dev.workflow.yml

```diff
@@ -3,44 +3,53 @@ name: Push to dev branch
 on:
   push:
     branches:
-      - 1-dev
-      - 2-dev
+      - "[0-9]+-dev"
 
 jobs:
   lint:
     name: Lint
     runs-on: ubuntu-22.04
     steps:
-      - name: Install SSH key
-        uses: shimataro/ssh-key-action@v2
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
         with:
-          key: ${{ secrets.SSH_KEY }}
-          name: id_rsa
-          known_hosts: unnecessary
+          node-version: 16
+      - name: Install dependencies
+        run: npm ci
+      - name: Run lint
+        run: npm run lint
+
+  check-types:
+    name: Check types
+    runs-on: ubuntu-22.04
+    steps:
       - uses: actions/checkout@v2
-      - uses: ./.github/actions/remove-libs
-      - uses: actions/setup-node@v1.4.4
+      - uses: actions/setup-node@v3
         with:
-          node-version: "14"
-      - uses: ./.github/actions/lint
+          node-version: 16
+      - name: Install dependencies
+        run: npm ci
+      - name: Run types tests
+        run: npm run test:types
 
   functional-tests:
     name: Functional Tests
     runs-on: ubuntu-22.04
-    needs: [lint]
+    needs: [lint, check-types]
     steps:
-      - name: Install SSH key
-        uses: shimataro/ssh-key-action@v2
-        with:
-          key: ${{ secrets.SSH_KEY }}
-          name: id_rsa
-          known_hosts: unnecessary
       - uses: actions/checkout@v2
-      - uses: ./.github/actions/remove-libs
-      - uses: actions/setup-node@v1.4.4
+      - uses: actions/setup-node@v3
         with:
-          node-version: "14"
-      - uses: ./.github/actions/functional-test
+          node-version: 16
+      - name: Install dependencies
+        run: npm ci
+      - name: Run tests
+        run: |
+          docker-compose up -d
+          # Start backend in background (& is mandatory to run process in background otherwise the task never continue)
+          npm run dev &
```

Comment:
Isn't the docker compose suppose to launch the whole task, why nom run dev is needed ?

Reply:
Docker only run redis and elasticsearch, kuzzle is run out of docker to improve the log of errors in CI.
Also maybe it's better to run `npm run prod` instead development mode for better tests.

Reply:
This is not a good idea, kuzzle should be ran inside the docker compose, because when it will be deployed in production it will be used inside a docker container, so when you test it you should run it in the same environment. else your environment will not be the same while running inside your CI, witch is not what you want

If you seem to be lacking of logs, you should add a docker-compose logs kuzzle when the script fails in the wait-nuzzle-start.sh

Reply:
The problem is not especially for `wait-kuzzle.sh` but when functional tests cause errors.
IMHO it's better like that, but no problem, I can restore into docker.

Reply:
I understand why you want it this way, but in this way we diverge too much from the final environment in which a kuzzle is deployed and this creates more risk of trouble. Maybe we can talk about it on discord a bit to see if we can still have a docker environment, and more log that suit the debugging of the stack when it runs functional test

wdyt ?